common/src/main/java/io/netty/util/internal/PlatformDependent.java

```diff
@@ -150,6 +156,102 @@
         }
         DIRECT_MEMORY_LIMIT = maxDirectMemory;
         logger.debug("io.netty.maxDirectMemory: {} bytes", maxDirectMemory);
+
+        _BITS = _Bits.getBits();
+    }
+
+    /**
+     * Util class to expose {@link java.nio.Bits} methods for
+     * direct memory management.
+     */
+    static class _Bits {
+
+        static final boolean pa = IS_DIRECT_MEMORY_PAGE_ALIGNED;
+        static final int ps = PlatformDependent0.UNSAFE.pageSize();
+
+        /** Hidden method to reserve memory */
+        Method reserveMemoryMethod;
+
+        /** Hidden method to unreserve memory */
+        Method unreserveMemoryMethod;
+
+        /**
+         * @return a new hook for direct memory management. It can be disabled
+         * by setting {@code -Dio.netty.nio.memory.tracking.enable=false}
+         */
+        static _Bits getBits() {
+            try {
+                if (!Boolean.parseBoolean(SystemPropertyUtil.get("io.netty.nio.memory.tracking.enable", "true"))) {
+                    return null;
+                }
+                return new _Bits();
+            } catch (Exception e) {
+                logger.warn("Failed to initialize nio hook.", e);
+                return null;
+            }
+        }
+
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Remove the declaration of thrown exception 'java.lang.SecurityException' which is a runtime exception. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://garage.netty.io/sonarqube/coding_rules#rule_key=squid%3ARedundantThrowsDeclarationCheck)